yellowbrick/classifier/threshold.py

```diff
@@ -0,0 +1,351 @@
+        YellowbrickTypeError exception on instantiation.
+
+    colormap : string or cmap, default: None
+        Optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+
+    color : String, default: None
+        Optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+
+    n_trials : Integer, default: 50
+        Number of trials to conduct via train_test_split
+
+    quantiles : Sequence, default: (0.1, 0.5, .9)
+        Setting the quantiles for visualizing model variability using
+        scipy.stats.mstats.mquantiles
+
+    random_state : Integer, default: 0
+        Random state integer for sampling in train_test_split
+
+    title : String, default: None
+        Title of the visualization
+
+    kwargs : keyword arguments passed to the super class.
+    """
+
+    def __init__(self,
+                 model,
+                 colormap=None,
+                 color=None,
+                 n_trials=50,
+                 test_size_percent=0.1,
+                 quantiles=(0.1, 0.5, 0.9),
+                 random_state=0,
+                 title=None,
+                 **kwargs):
+        # Check to see if model is an instance of a classifier.
+        # Should return an error if it isn't.
+        if not isclassifier(model):
+            raise YellowbrickTypeError(
+                "This estimator is not a classifier; try a regression or clustering score visualizer instead!"
+            )
+        super(ThresholdVisualizer, self).__init__(model, **kwargs)
+
+        self.estimator = model
+        self.colormap = kwargs.pop('colormap', None)
+        self.color = kwargs.pop('color', None)
+        self.title = title
+        self.n_trials = n_trials
+        self.test_size_percent = test_size_percent
+        self.quantiles = quantiles
+        self.random_state = np.random.RandomState(random_state)
+
+        # to be set later
+        self.plot_data = None
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the visualizer
+        """
+        self.plot_data = []
+
+        for trial in range(self.n_trials):
+            train_X, test_X, train_y, test_y = train_test_split(
+                X,
+                y,
+                test_size=self.test_size_percent,
+                random_state=self.random_state)
```

Comment:
okay, set the default to None (no set random state unless user defines it in init)